build failure: error: use of deleted function ‘{anonymous}::Block::Block(const {anonymous}::Block&)’

at b029f50
My previous successful build was 0.5.0-dev+4791 at 74b3042
```
AWS-Sachs-Ubuntu$ make
    CC src/cgmemmgr.o
/home/sachs/src/julia-master/src/cgmemmgr.cpp: In function ‘void {anonymous}::write_self_mem(void*, void*, size_t)’:
/home/sachs/src/julia-master/src/cgmemmgr.cpp:282:20: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (ret == size)
                    ^
In file included from /home/sachs/src/julia-master/usr/include/llvm/ExecutionEngine/SectionMemoryManager.h:18:0,
                 from /home/sachs/src/julia-master/src/cgmemmgr.cpp:8:
/home/sachs/src/julia-master/usr/include/llvm/ADT/SmallVector.h: In instantiation of ‘static void llvm::SmallVectorTemplateBase<T, isPodLike>::uninitialized_move(It1, It1, It2) [with It1 = {anonymous}::Block*; It2 = {anonymous}::Block*; T = {anonymous}::Block; bool isPodLike = false]’:
/home/sachs/src/julia-master/usr/include/llvm/ADT/SmallVector.h:254:3:   required from ‘void llvm::SmallVectorTemplateBase<T, isPodLike>::grow(size_t) [with T = {anonymous}::Block; bool isPodLike = false; size_t = long unsigned int]’
/home/sachs/src/julia-master/usr/include/llvm/ADT/SmallVector.h:654:7:   required from ‘void llvm::SmallVectorImpl<T>::emplace_back(ArgTypes&& ...) [with ArgTypes = {}; T = {anonymous}::Block]’
/home/sachs/src/julia-master/src/cgmemmgr.cpp:597:32:   required from ‘void* {anonymous}::SelfMemAllocator<exec>::get_wr_ptr({anonymous}::SplitPtrBlock&, void*, size_t, size_t) [with bool exec = true; size_t = long unsigned int]’
/home/sachs/src/julia-master/src/cgmemmgr.cpp:859:1:   required from here
/home/sachs/src/julia-master/usr/include/llvm/ADT/SmallVector.h:206:7: error: use of deleted function ‘{anonymous}::Block::Block(const {anonymous}::Block&)’
       ::new ((void*) &*Dest) T(::std::move(*I));
       ^
/home/sachs/src/julia-master/src/cgmemmgr.cpp:335:5: error: declared here
     Block(const Block&) = delete;
     ^
/home/sachs/src/julia-master/src/cgmemmgr.cpp: In function ‘bool {anonymous}::check_fd_or_close(int)’:
/home/sachs/src/julia-master/src/cgmemmgr.cpp:108:32: warning: ignoring return value of ‘int ftruncate(int, __off_t)’, declared with attribute warn_unused_result [-Wunused-result]
     ftruncate(fd, jl_page_size);
                                ^
make[1]: *** [cgmemmgr.o] Error 1
make: *** [julia-src-release] Error 2
```

--- a/src/cgmemmgr.cpp
+++ b/src/cgmemmgr.cpp
@@ -15,9 +15,6 @@
 #endif
 #ifdef _OS_LINUX_
 #  include <sys/syscall.h>
-#  ifdef __NR_memfd_create
-#    include <linux/memfd.h>
-#  endif
 #endif
 #ifndef _OS_WINDOWS_
 #  include <sys/mman.h>
@@ -102,10 +99,13 @@ static bool check_fd_or_close(int fd)
 {
     if (fd == -1)
         return false;
-    // This can fail due to `noexec` mount option ....
     fcntl(fd, F_SETFD, FD_CLOEXEC);
     fchmod(fd, S_IRWXU);
-    ftruncate(fd, jl_page_size);
+    if (ftruncate(fd, jl_page_size) != 0) {
+        close(fd);
+        return false;
+    }
+    // This can fail due to `noexec` mount option ....
     void *ptr = mmap(nullptr, jl_page_size, PROT_READ | PROT_EXEC,
                      MAP_SHARED, fd, 0);
     if (ptr == MAP_FAILED) {
@@ -159,7 +159,7 @@ static intptr_t get_anon_hdl(void)
     // Linux and FreeBSD can create an anonymous fd without touching the
     // file system.
 #  ifdef __NR_memfd_create
-    fd = syscall(__NR_memfd_create, "julia-codegen", MFD_CLOEXEC);
+    fd = syscall(__NR_memfd_create, "julia-codegen", 0);
     if (check_fd_or_close(fd))
         return fd;
 #  endif
@@ -279,7 +279,7 @@ static void write_self_mem(void *dest, void *ptr, size_t size)
 {
     while (size > 0) {
         ssize_t ret = pwrite(self_mem_fd, ptr, size, (uintptr_t)dest);
-        if (ret == size)
+        if ((size_t)ret == size)
             return;
         if (ret == -1 && (errno == EAGAIN || errno == EINTR))
             continue;
@@ -334,6 +334,14 @@ struct Block {
 
     Block(const Block&) = delete;
     Block &operator=(const Block&) = delete;
+    Block(Block &&other)
+        : ptr(other.ptr),
+          total(other.total),
+          avail(other.avail)
+    {
+        other.ptr = nullptr;
+        other.total = other.avail = 0;
+    }
 
     Block() = default;
 

